### ISSUE TITLE
[BUDI-7143] Formula column on external database clears after save

### ISSUE THREAD

**Issue by shawnweeks** [Budibase/budibase#10670]
**Checklist**

* [X] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**

* Self
  * Method: docker compose
  * Budibase Version: 2.6.20
  * App Version: 2.6.20

**Describe the bug**
A formula column against a link/relationship column in an external database table cannot be edited after you save it and is blanked out after calling fetch table. In my example I have a very simple `parent` and `child` table and I use a formula like `{{ length child }}` to get a count of child rows. A seen in the screenshots this formula blanks out after saving and if you call a fetch tables it quits calculating. If you just save it and don't touch it or fetch table then it will continue working.

**To Reproduce**

1. Create the tables listed below in a Postgres or other compatible database.
2. Add the data source to Budibase and import tables.
3. Create a one to many relationship between `parent` and `child` using `id` on `parent` and `parent_id` on `child`
4. In the `parent` table create a formula column using the `child` link column, any formula will do it but I used `{{ length child }}`
5. Save the new column as `child_count` and close out of the column editor.
6. Go back to column editor and notice the formula is now just `0`

```
create table parent (
    id int generated always as identity primary key,
    name text not null
);

create table child (
    id int generated always as identity primary key,
    parent_id int not null,
    name text not null
);
```

**Expected behavior**
Formulas should be retained after saving column and fetching tables from database.

**Screenshots**

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/587a865e-f1bb-418b-a485-29db7db4e7c8/af06a0c5-cfb8-4cb7-83d2-a8af09214f07)

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/f9a26938-4b6a-4aac-9243-0b2bbd60007f/6f3b4741-c2d3-44b4-b1b3-605bfc6bd0f2)

**App Export**
[External Table Formula Bug-export-1684848670415.tar.gz](https://github.com/Budibase/budibase/files/11544536/External.Table.Formula.Bug-export-1684848670415.tar.gz)

**Desktop (please complete the following information):**

* OS: MacOS
* Browser Firefox and Chrome
* Version Latest

**Additional context**
I recognize that this is probably related to [https://github.com/Budibase/budibase/issues/10193](https://github.com/Budibase/budibase/issues/10193) or even the same issue but I noticed that more was going on than that ticket described and it's just not MariaDB. I've confirmed MySQL does the exact same thing so I don't think it's external database specific.

<sub>From [SyncLinear.com](https://synclinear.com) | [BUDI-7143](https://linear.app/budibase/issue/BUDI-7143/formula-against-relationship-column-on-external-database-clears-after)</sub>

**Comment by melohagan** [comment]
This seems to be specific to `length` from what I can tell so far.

**Comment by shawnweeks** [comment]
@melohagan It's not just limited to array columns. I created a formula column that is just the JavaScript `return 10*10;` and it's also won't let you edit it after saving.

<img width="1024" alt="Screenshot 2023-06-16 at 11 02 11 AM" src="https://github.com/Budibase/budibase/assets/9768773/a050fdb4-6923-4bfb-9352-9e162876359b">

<img width="418" alt="Screenshot 2023-06-16 at 11 02 36 AM" src="https://github.com/Budibase/budibase/assets/9768773/9e069ff0-b17e-40f8-92a2-ffb78c8f6ce7">

<img width="413" alt="Screenshot 2023-06-16 at 11 02 45 AM" src="https://github.com/Budibase/budibase/assets/9768773/807d153f-94e1-4ba3-8a04-2c68a50e4541">

**Comment by melohagan** [comment]
Fixed in **2.7.30**
